docs: add release notes for 0.28.1 (#8861)
determined-ai/determined: docs/release-notes.rst

 Version 0.28
**************

Version 0.28.1
==============

**Release Date:** February 20, 2024

**Improvements**

> -  GCS Storage client will now retry following the default policy on ``TooManyRequests`` rate limit

Reviewer: ```suggestion
-  The Google Cloud Storage client will now retry following the default policy on ``TooManyRequests`` rate limit
```

Reviewer 2: good catch

Reviewer 2: updated the usage guide https://hpe-aiatscale.atlassian.net/wiki/spaces/DOC/pages/1332707345/Usage+Guide

   errors.

**Bug Fixes**

> -  Since 0.26.2, it was possible to cause Determined Trials and Commands to hang after the main

Reviewer: ```suggestion
-  Since 0.26.2, it was possible to cause Determined trials and commands to hang after the main
```

Reviewer 2: updated the usage guide https://hpe-aiatscale.atlassian.net/wiki/spaces/DOC/pages/1332707345/Usage+Guide

   process exited but before the container exited, by starting a non-terminating subprocess from
   your training script or command that kept an open stdout or stderr file descriptor. Now, logs

>    from subprocesses of your main process are ignored after your main process has exited.

Reviewer 2: `stdout` or `stderr` ???

Reviewer 2: in rst it would be ``stdout`` or ``stderr``

Author: Updated

> -  TensorBoard: For TensorBoard task types, fix a bug that would allow users to view TensorBoards,
>       even if they did not have permission to view the all workspaces it included.

Reviewer: Not quite sure what this is trying to say, but perhaps:
```suggestion
-  TensorBoard: Fix a bug that would allow users to view TensorBoards even if they did not have
   permission to view the corresponding workspaces.
```
